### DISCUSSION TITLE
Uploading partially labelled datasets

### DISCUSSION THREAD

**Discussion by wouterharmsen** [asreview/asreview#1237] [Q&A] [answered]
Dear community,

Anyone an idea of uploading partially labelled datasets in ASReview? 
The labelled part is our prior knowledge before starting screening with ASReview.

Hope to hear fom you.
Wouter

**Comment by gimoAI** [comment, marked as answer]
Hi @wouterharmsen,

If you enter Oracle mode and upload your partially labeled dataset, ASReview recognizes the labeled data as prior knowledge :smiley:

Just make sure your data is labeled properly for ASReview, see [data format](https://asreview.readthedocs.io/en/latest/data_format.html).

**Reply by miriamvdmaten** [reply]
Thanks @gimoAI ! Just wondering, would it also be possible to upload two separate datafiles (one labeled, one not) which will then be merged into one project? Or add new data to a current project? 

Thanks in advance!

**Reply by gimoAI** [reply]
Hi @miriamvdmaten! 

I'm currently writing a "combine" script for exactly this: combining different datafiles such as labeled, partly labeled, unlabeled etc. It shouldn't be too long till you can access it :smile:

**Reply by miriamvdmaten** [reply]
@gimoAI that's perfect! And happy to hear that you got involved in the project from the ASReview side :)

**Comment by gimoAI** [comment]
@wouterharmsen @miriamvdmaten, I would like to bring to your attention that the [ASReview datatools extension](https://github.com/asreview/asreview-datatools) has a new feature "[compose](https://github.com/asreview/asreview-datatools#data-compose-experimental)" to turn multiple datasets into a single (labeled, partly labeled, or unlabeled) dataset.

**Reply by miriamvdmaten** [reply]
Perfect @gimoAI. We'll have a look at it

**Reply by nwilts** [reply]
Hi all, I have a question about the "compose" extension. I'm currently trying to combine three partially labeled datasets (.ris format) into one using the compose extension, but it only includes one dataset out of the three I'm trying to combine. I'm currently using the code: asreview data compose composed_data.ris -l anxiety.ris -l depression.ris -l SUD.ris --hierarchy riu, and it is only including the SUD.ris dataset and disregarding the others. If I change the -l to -r and -i for anxiety and depression respectively it is combining all the datasets into one, so there appears to be an issue with having multiple datasets with the same -l code. How can I solve this?

**Reply by Rensvandeschoot** [reply]
It seems you aim to combine three datasets, all with labeled records, right? If so, you should use the `vstack`  function.
